OneLauncher/Settings.py

```diff
@@ -275,13 +278,15 @@ def SaveSettings(self, saveAccountDetails=None, savePassword=None, game=None):
             tempNode.appendChild(doc.createTextNode("False"))
         gameConfigNode.appendChild(tempNode)
 
-        tempNode = doc.createElementNS(EMPTY_NAMESPACE, "x64Client")
-        if self.x64ClientEnabled:
-            tempNode.appendChild(doc.createTextNode("True"))
-        else:
-            tempNode.appendChild(doc.createTextNode("False"))
+        tempNode = doc.createElementNS(EMPTY_NAMESPACE, "Client")
+        tempNode.appendChild(doc.createTextNode("%s" % (self.client)))
         gameConfigNode.appendChild(tempNode)
 
+        if self.client == "WIN32":
+            tempNode = doc.createElementNS(EMPTY_NAMESPACE, "x64Client")
+            tempNode.appendChild(doc.createTextNode("False"))
+            gameConfigNode.appendChild(tempNode)
```

Comment:
```suggestion
        if self.client in ["WIN32", "WIN64"]:
            tempNode = doc.createElementNS(EMPTY_NAMESPACE, "x64Client")
            value = "True" if self.client == "WIN64" else "False"
            tempNode.appendChild(doc.createTextNode(value))
            gameConfigNode.appendChild(tempNode)
```

Sorry. I wasn't clear in my other comment on this. I just meant that x64Client doesn't have to do with the legacy client, so it should only be false if the normal 32-bit client is in use.